third_party/amd/lib/TritonAMDGPUToLLVM/Utility.h

```diff
@@ -112,6 +112,24 @@ bool isChainDotHead(mlir::triton::DotOpInterface dotOp);
 // in the same region
 bool isChainDotTail(mlir::triton::DotOpInterface dotOp);
 
+// Determine the order in which CTA tiles are laid out across the tensor.
+// That is, create vector of dimensions from fastest to slowest varying.
+SmallVector<unsigned> getCTATileOrder(MLIRContext *ctx,
+                                      const LinearLayout &layout);
+
+template <typename T, typename U, typename BinaryOp>
+std::vector<unsigned> multiDimElementwise(const ArrayRef<T> &lhs,
```

Comment:
To share this function among `TritonAMDGPUIR` and `TritonAMDGPUDialectToLLVM` it's better to put this function in some neutral place like `third_party/amd/lib/Utils`.